test/tryGetDefaultBranch.test.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import assert = require('assert');
+import { Uri } from 'vscode';
+import { cpUtils, getGitWorkspaceState, GitWorkspaceState, tryGetDefaultBranch } from '../extension.bundle';
+import { createTestActionContext, testFolderPath } from './global.test';
+
+suite('Get default branch for Git repo', function (this: Mocha.Suite): void {
+    this.timeout(30 * 1000);
+    const context = createTestActionContext();
+    const testFolderUri: Uri = Uri.file(testFolderPath);
+    const localDefaultBranch: string = 'localDefault';
+    const globalDefaultBranch: string = 'globalDefault';
+
+    test('Workspace with default branch set in global config', async () => {
+        await cpUtils.executeCommand(undefined, undefined, 'git', 'config', '--global', 'init.defaultBranch', globalDefaultBranch);
+        const gitWorkspaceState: GitWorkspaceState = await getGitWorkspaceState(context, testFolderUri);
+
+        if (!gitWorkspaceState.repo) {
+            throw new Error('Could not retrieve git repository.');
+        }
+
+        await gitWorkspaceState.repo.createBranch(globalDefaultBranch, false);
+        assert.strictEqual(await tryGetDefaultBranch(gitWorkspaceState.repo), globalDefaultBranch);
+    });
+
+    test('Workspace with default branch set in local config', async () => {
+        await cpUtils.executeCommand(undefined, testFolderPath, 'git', 'config', '--local', 'init.defaultBranch', localDefaultBranch);
```

Comment:
This is more related to the production code over the test itself, but I would not expect "init.defaultBranch" to be the main determiner of the default branch. I would expect it to be determined from the remote. For example, my default branch might be "master" in my local git config, but my default branch for all of our extension repos would be "main" by this point. I did a bit of googling and I'm not sure the best way to get that info, but here's one example:

https://stackoverflow.com/questions/28666357/git-how-to-get-default-branch

Reply:
I think that's fair.  The `init.defaultBranch` setting is relatively new (https://github.blog/2020-07-27-highlights-from-git-2-28/).  I'll put up a PR that will use the remote defaultBranch if it exists, otherwise fallback to this local setting.